loopd/daemon.go

```diff
@@ -171,34 +176,8 @@ func (d *Daemon) Run() error {
 		log.Infof("REST proxy disabled")
 	}
 
-	mainCtx, cancel := context.WithCancel(context.Background())
-
-	// Start the swap client itself.
-	d.wg.Add(1)
-	go func() {
-		defer d.wg.Done()
-
-		log.Infof("Starting swap client")
-		err := d.impl.Run(mainCtx, d.statusChan)
-		if err != nil {
-			log.Error(err)
-		}
-		log.Infof("Swap client stopped")
-
-		log.Infof("Stopping gRPC server")
-		grpcServer.Stop()
-
-		cancel()
-	}()
-
-	// Start a goroutine that broadcasts swap updates to clients.
-	d.wg.Add(1)
-	go func() {
-		defer d.wg.Done()
-
-		log.Infof("Waiting for updates")
-		d.processStatusUpdates(mainCtx)
-	}()
+	// Start the main event handlers.
+	d.startEventHandlers()
```

Comment:
The name 'event handlers' maybe doesn't do right to the core swap logic that is started in this function